remotes/fixup.go

```diff
@@ -204,47 +204,94 @@ func fixupBaseImage(ctx context.Context, name string, baseImage *bundle.BaseImag
 		return imageFixupInfo{}, false, err
 	}
 
-	var (
-		descriptor     ocischemav1.Descriptor
-		pushed         bool
-		sourceImageRef reference.Named
-	)
-
-	if baseImage.Image == "" && cfg.pushImages {
-		// no image is defined for the service, try to push by digest from local docker image store
-		descriptor, err = pushImageToTarget(ctx, baseImage.Digest, cfg)
+	fixups := []func(context.Context, reference.Named, *bundle.BaseImage, fixupConfig) (imageFixupInfo, bool, bool, error){
+		pushByDigest,
+		resolveImage,
+		resolveImageInRelocationMap,
```

Comment:
I think we should resolve in relocation map first 🤔

Reply:
Updated.
New order is then:

- digest if no other information
- image from relocation map
- image from bundle
- local image

The test case has been updated to show many cases and order.

Reply:
Cool, thanks 👍 
WDYT @radu-matei ?
I think we should always push the image found in the relocation map first. Eg: someone pulls a bundle using `nginx` as service image. When re-pushing this bundle, it should push the digested reference found in the relocation map, aka the service image found in the repository along the bundle, not the actual `nginx` image.